packages/@lwc/template-compiler/src/__tests__/fixtures/scoped-slots/invalid/slot-data/with-for-each-parent/metadata.json

```diff
@@ -0,0 +1,37 @@
+{
+    "warnings": [
+        {
+            "code": 1178,
+            "message": "LWC1178: Invalid usage of 'lwc:slot-data' in an iterator(for:each, for:of) block.",
+            "level": 1,
+            "location": {
+                "line": 5,
+                "column": 13,
+                "start": 154,
+                "length": 125
+            }
+        },
+        {
+            "code": 1178,
+            "message": "LWC1178: Invalid usage of 'lwc:slot-data' in an iterator(for:each, for:of) block.",
+            "level": 1,
+            "location": {
+                "line": 12,
+                "column": 13,
+                "start": 404,
+                "length": 182
+            }
+        },
```

Comment:
> lwc:slot-data is not the direct child of the component <x-list>.

That is not an issue. For example, the following is valid:
```
<template>
     <x-list>
           <template lwc:if={flag}>
                  <template lwc:slot-data="item">
                          <spam>{item.id}</span>
                  </template>
           </template>
     </x-list>
</template>
```

The restriction is specifically about having an iteration block around scoped slot content. It smells of an anti-pattern. Conceptually, scoped slot was created so that the child does the iteration etc and gets a fragment from the parent to reuse. This is for the same reason that duplicate slot content are also restricted(`NO_DUPLICATE_SCOPED_SLOT`).
If a parent produces multiple fragment factories in an iteration, the child has to invoke each of the factories with its data and slot them in. This will result in a O(n2) _O-of-n-squared_ complexity.